### ISSUE TITLE
Error: "Failed to report coverage! Error: No coverage results collected."

### ISSUE THREAD

**Issue by adder46** [xd009642/tarpaulin#633]
I wanted to try out tarpaulin on my [project,](https://github.com/adder46/adder) however, I'm getting an error:
```
$ cargo tarpaulin --debug
Nov 29 19:00:04.813 DEBUG cargo_tarpaulin: set up logging
Nov 29 19:00:04.813  INFO cargo_tarpaulin::config: Creating config
Nov 29 19:00:04.813 TRACE cargo_tarpaulin: Debug mode activated
Nov 29 19:00:04.827  INFO cargo_tarpaulin: Running Tarpaulin
Nov 29 19:00:04.827  INFO cargo_tarpaulin: Building project
Nov 29 19:00:04.841 TRACE cargo_tarpaulin::cargo: Setting RUSTDOCFLAGS=' -C relocation-model=dynamic-no-pic -C link-dead-code -C debuginfo=2 --cfg=tarpaulin  --persist-doctests /home/alex/.repos/adder/target/doctests -Z unstable-options '
Nov 29 19:00:04.841 TRACE cargo_tarpaulin::cargo: Running command "cargo" "+stable-x86_64-unknown-linux-gnu" "test" "--no-run" "--message-format" "json" "--manifest-path" "/home/alex/.repos/adder/Cargo.toml" "--tests" "-vvv" "--target-dir" "/home/alex/.repos/adder/target"
       Fresh adder v0.1.0 (/home/alex/.repos/adder)
    Finished test [unoptimized + debuginfo] target(s) in 0.01s
Nov 29 19:00:04.860 TRACE cargo_tarpaulin::source_analysis: Branch analysis
Nov 29 19:00:04.860 TRACE cargo_tarpaulin::source_analysis: {}
Nov 29 19:00:04.877  INFO cargo_tarpaulin: Launching test
Nov 29 19:00:04.877  INFO cargo_tarpaulin: running /home/alex/.repos/adder/target/debug/deps/adder-dae29a78559bde73
Nov 29 19:00:05.043 TRACE Collect coverage{pid=1445800}: cargo_tarpaulin::statemachine::linux: Caught inferior transitioning to Initialise state

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

Nov 29 19:00:05.044 TRACE Collect coverage{pid=1445800}: cargo_tarpaulin::statemachine::linux: Initialised inferior, transitioning to wait state
Nov 29 19:00:05.044 TRACE Collect coverage{pid=1445800}: cargo_tarpaulin::statemachine::linux: Result queue is [PtraceEvent(Pid(1445800), SIGTRAP, 6)]
Nov 29 19:00:05.044 TRACE Collect coverage{pid=1445800}: cargo_tarpaulin::statemachine::linux: Child exiting
Nov 29 19:00:05.044 TRACE Collect coverage{pid=1445800}: cargo_tarpaulin::statemachine::linux: Result queue is [Exited(Pid(1445800), 0)]
Nov 29 19:00:05.044 TRACE Collect coverage{pid=1445800}: cargo_tarpaulin::statemachine::linux: Exited Pid(1445800) parent Pid(1445800)
Nov 29 19:00:05.044 TRACE Collect coverage{pid=1445800}: cargo_tarpaulin::statemachine::linux: No action suggested to continue tracee. Attempting a continue
Nov 29 19:00:05.057  INFO cargo_tarpaulin: Launching test
Nov 29 19:00:05.057  INFO cargo_tarpaulin: running /home/alex/.repos/adder/target/debug/deps/test_ripplecarry-8e391c50091f1c4c
Nov 29 19:00:05.225 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Caught inferior transitioning to Initialise state

running 3 tests
Nov 29 19:00:05.226 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Initialised inferior, transitioning to wait state
Nov 29 19:00:05.226 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [PtraceEvent(Pid(1445801), SIGTRAP, 3)]
Nov 29 19:00:05.226 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: New thread spawned 1445802
Nov 29 19:00:05.226 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [PtraceEvent(Pid(1445801), SIGTRAP, 3)]
Nov 29 19:00:05.226 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: New thread spawned 1445803
Nov 29 19:00:05.226 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [PtraceEvent(Pid(1445801), SIGTRAP, 3)]
Nov 29 19:00:05.226 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: New thread spawned 1445804
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [Stopped(Pid(1445804), SIGSTOP), Stopped(Pid(1445803), SIGSTOP), Stopped(Pid(1445802), SIGSTOP)]
test add_overflow ... ok
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [PtraceEvent(Pid(1445804), SIGTRAP, 6)]
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Child exiting
test add_max ... ok
test add ... ok

test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [PtraceEvent(Pid(1445801), SIGTRAP, 6)]
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Child exiting
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [Exited(Pid(1445804), 0)]
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Exited Pid(1445804) parent Pid(1445801)
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [Exited(Pid(1445803), 0)]
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Exited Pid(1445803) parent Pid(1445801)
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [PtraceEvent(Pid(1445802), SIGTRAP, 6)]
Nov 29 19:00:05.233 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Child exiting
Nov 29 19:00:05.238 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [Exited(Pid(1445802), 0)]
Nov 29 19:00:05.238 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Exited Pid(1445802) parent Pid(1445801)
Nov 29 19:00:05.238 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Result queue is [Exited(Pid(1445801), 0)]
Nov 29 19:00:05.238 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: Exited Pid(1445801) parent Pid(1445801)
Nov 29 19:00:05.238 TRACE Collect coverage{pid=1445801}: cargo_tarpaulin::statemachine::linux: No action suggested to continue tracee. Attempting a continue
Nov 29 19:00:05.238  INFO cargo_tarpaulin::event_log: Serializing tarpaulin debug log to tarpaulin-run-2020-11-29T19:00:05.238125677+01:00.json
Error: "Failed to report coverage! Error: No coverage results collected."
```

**Comment by xd009642** [comment]
That's really odd, I can't recreate it with the current dev version or the latest version I can install from crates io. Which version of tarpaulin are you on? 0.17 was yanked so if it's that then that might explain it.

**Comment by adder46** [comment]
```
$ cargo tarpaulin --version
cargo-tarpaulin version: 0.16.0
```
I installed it with `cargo install cargo-tarpaulin`.

How should I install tarpaulin?

**Comment by xd009642** [comment]
That's correct, version and installation. Hmm what version of rust are you using and what OS?

**Comment by adder46** [comment]
Xubuntu 20.04 LTS, Rust v1.48.0.

**Comment by xd009642** [comment]
okay, same rust as me but different Ubuntu. I'll let you know when I have any progress. In the meantime you could always run via CI and use something like coveralls or codecov to view the results

**Comment by adder46** [comment]
I've tried to [move the tests](https://github.com/adder46/adder/blob/tests/src/lib.rs#L95). 

And the result:

```
$ cargo tarpaulin
Nov 29 20:36:53.515  INFO cargo_tarpaulin::config: Creating config
Nov 29 20:36:53.533  INFO cargo_tarpaulin: Running Tarpaulin
Nov 29 20:36:53.533  INFO cargo_tarpaulin: Building project
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling version_check v0.9.2
   Compiling syn v1.0.52
   Compiling test-case v1.0.0
   Compiling quote v1.0.7
   Compiling adder v0.1.0 (/home/alex/.repos/adder)
    Finished test [unoptimized + debuginfo] target(s) in 9.94s
Nov 29 20:37:03.516  INFO cargo_tarpaulin: Launching test
Nov 29 20:37:03.516  INFO cargo_tarpaulin: running /home/alex/.repos/adder/target/debug/deps/adder-a67d185ca3864d02

running 11 tests
test tests::full_add::_1_and_1 ... ok
test tests::full_add::_1_and_0 ... ok
test tests::full_add::_0_and_1 ... ok
test tests::half_add::_0_and_0 ... ok
test tests::full_add::_0_and_0 ... ok
test tests::ripple_add ... ok
test tests::half_add::_1_and_1 ... ok
test tests::ripple_add_max ... ok
test tests::ripple_add_overflow ... ok
test tests::half_add::_1_and_0 ... ok
test tests::half_add::_0_and_1 ... ok

test result: ok. 11 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

Error: "Failed to report coverage! Error: No coverage results collected."

```

**Comment by xd009642** [comment]
So while revisiting open issues I tried tarpaulin on your project and got

```
running 11 tests
test tests::half_add::case_4 ... ok
test tests::half_add::case_3 ... ok
test tests::half_add::case_2 ... ok
test tests::half_add::case_1 ... ok
test tests::full_add::case_4 ... ok
test tests::full_add::case_1 ... ok
test tests::full_add::case_3 ... ok
test tests::full_add::case_2 ... ok
test tests::ripple_add::case_3 ... ok
test tests::ripple_add::case_2 ... ok
test tests::ripple_add::case_1 ... ok

test result: ok. 11 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.03s

Feb 19 15:14:06.588  INFO cargo_tarpaulin::report: Coverage Results:
|| Tested/Total Lines:
|| src/lib.rs: 53/55
|| 
96.36% coverage, 53/55 lines covered
```

And I also see you have a codecov badge on the repo (though I can't see where coverage is collected). I'm just going to close this as it looks to be solved though feel free to reopen it if I'm mistaken.

**Comment by adder46** [comment]
Yeah, it started to work when I reinstalled the complete system (not because of tarpaulin). Then the Github actions docker image started failing on all my projects but I was too lazy to report the issue.

**Comment by xd009642** [comment]
Ah yeah I realised that a yanked release was the image in latest tag. That's been fixed by bumping latest to the 0.18.0 alpha though if you want to try it again
